datafusion/optimizer/src/decorrelate_general.rs

```diff
@@ -2300,29 +2744,156 @@ Projection: outer_table.a, outer_table.b, outer_table.c [a:UInt32, b:UInt32, c:U
+                              Left Join:  Filter: inner_table_lv1.b IS NOT DISTINCT FROM delim_scan_4.b [a:UInt32, b:UInt32, c:UInt32, a:UInt32;N, c:UInt32;N, CASE WHEN count(inner_table_lv2.a) IS NULL THEN Int32(0) ELSE count(inner_table_lv2.a) END:Int32;N, c:UInt32;N, a:UInt32;N, b:UInt32;N, b:UInt32;N, a:UInt32;N, c:UInt32;N]
                                 Cross Join:  [a:UInt32, b:UInt32, c:UInt32, a:UInt32;N, c:UInt32;N]
                                   TableScan: inner_table_lv1 [a:UInt32, b:UInt32, c:UInt32]
                                   SubqueryAlias: delim_scan_2 [a:UInt32;N, c:UInt32;N]
                                     DelimGet: outer_table.a, outer_table.c [a:UInt32;N, c:UInt32;N]
-                                Projection: CASE WHEN count(inner_table_lv2.a) IS NULL THEN Int32(0) ELSE count(inner_table_lv2.a) END, delim_scan_4.a, delim_scan_4.c, delim_scan_4.b, delim_scan_3.b, delim_scan_3.c, delim_scan_3.a [CASE WHEN count(inner_table_lv2.a) IS NULL THEN Int32(0) ELSE count(inner_table_lv2.a) END:Int32, a:UInt32;N, c:UInt32;N, b:UInt32;N, b:UInt32;N, c:UInt32;N, a:UInt32;N]
-                                  Inner Join:  Filter: inner_table_lv1.b IS NOT DISTINCT FROM delim_scan_3.b AND outer_table.c IS NOT DISTINCT FROM delim_scan_3.c AND outer_table.a IS NOT DISTINCT FROM delim_scan_3.a [count(inner_table_lv2.a):Int64, a:UInt32;N, c:UInt32;N, b:UInt32;N, b:UInt32;N, c:UInt32;N, a:UInt32;N]
-                                    Projection: count(inner_table_lv2.a), delim_scan_4.a, delim_scan_4.c, delim_scan_4.b [count(inner_table_lv2.a):Int64, a:UInt32;N, c:UInt32;N, b:UInt32;N]
-                                      Aggregate: groupBy=[[delim_scan_4.b, delim_scan_4.c, delim_scan_4.a]], aggr=[[count(inner_table_lv2.a)]] [b:UInt32;N, c:UInt32;N, a:UInt32;N, count(inner_table_lv2.a):Int64]
-                                        Filter: inner_table_lv2.a = delim_scan_4.a AND inner_table_lv2.b = delim_scan_4.b [a:UInt32, b:UInt32, c:UInt32, b:UInt32;N, c:UInt32;N, a:UInt32;N]
-                                          Cross Join:  [a:UInt32, b:UInt32, c:UInt32, b:UInt32;N, c:UInt32;N, a:UInt32;N]
+                                Projection: CASE WHEN count(inner_table_lv2.a) IS NULL THEN Int32(0) ELSE count(inner_table_lv2.a) END, delim_scan_4.c, delim_scan_4.a, delim_scan_4.b, delim_scan_3.b, delim_scan_3.a, delim_scan_3.c [CASE WHEN count(inner_table_lv2.a) IS NULL THEN Int32(0) ELSE count(inner_table_lv2.a) END:Int32, c:UInt32;N, a:UInt32;N, b:UInt32;N, b:UInt32;N, a:UInt32;N, c:UInt32;N]
```

Comment:
The projection list contains a lot of duplicate column after the join (delim_scan_4.c and delim_scan3.c are basically equivalent), we need to add dedup logic later (now i understand why the paper explicitly mention that they only keep 1 of the column after delim join)
And also the way we name the column after delim_scan will fail if 2 table having column with the same name (t1.a and t2.a being delim_scan will provides 2 duplicate column name delim_scan_$id.a)